lib/shopify_api/only_sync_dirty_attributes.rb

```diff
@@ -0,0 +1,34 @@
+module ShopifyAPI
+  module OnlySyncDirtyAttributes
+    def method_missing(method_symbol, *args)
+      method_name = method_symbol.to_s
+
+      if match_data = /(?<attribute_name>.+)(?:=)$/.match(method_name)
+        changed << match_data[:attribute_name]
+      end
+
+      super
+    end
+
+    def save
+      slice_attributes_on_changed
```

Comment:
`ensure @changed.clear` wouldn't work because we dont want to clear our dirty state if the request did not succeed (did not throw an exception). I like the idea otherwise